### ISSUE TITLE
Implement context responses that help describe transitions of tilt and area changes due to chaging angles and changing side length

### ISSUE THREAD

**Issue by terracoda** [phetsims/quadrilateral#154]
Following up to basic context responses fro finding and losing a parallelogram (#143), and responses that call out equality (#144), we need some transitory context responses about changing tilt (angle changes) and changing area of shape size (side length changes).

There are two tables in the design doc. One that outlines examples for maintaining a parallelogram and one that outlines examples for maintaining shapes:
- [Table: CWT: TMQ Transformations that Maintain a Parallelogram](https://docs.google.com/document/d/1jXayebAWnnNzsT3l6o72YPw4-YtiQaHQNuAi64eiguc/edit#heading=h.5e8ms9xel9zi)
- [Table: CWT: TMQ Transformations that Loose a Parallelogram & Make other Shapes](https://docs.google.com/document/d/1jXayebAWnnNzsT3l6o72YPw4-YtiQaHQNuAi64eiguc/edit#heading=h.pdkj2ec5sovz) 

@jessegreenberg, you can get started on these. I am just double checking the phrase for the non-parallelograms.

**Comment by terracoda** [comment]
**Here's the current phrase logic:**
- Proximity to Parallel 
    - always in parallel when maintaining a parallelogram.
    - always in some proximity to parallel when NOT maintaining a parallelogram.

For parallelograms the first part of response describes tilt changes in 3 ways
- Tilt changes in Parallel
    - Opposite sides {{in parallel}} +
    - Opposite sides {{tilt}} {{in parallel}} +
    - Opposite sides {{straighten}} {{in parallel}} +

For non-parallelograms the first part of response describes proximity to parallel
- Tilt changes outside of a parallelogram
- When 1 pair of sides are in parallel
    - Side {{sideName}} and {{sideName}} tilt {{proximityToParallel}} + 
- When no sides are in parallel:
    - {{All}} sides tilt {{away from}} parallel +
- When both sides move into parallel: 
    - {{Found a parallelogram}}.


Second part of response describes corner or area change
- Mostly a tilt change describe corners: 
          + as {{opposite}} corners change {{equally/unequally}}
- Mostly an area change (side length change): 
          + as shape gets {{smaller/bigger}}}.
          - AND maybe AREA can be used for combined changes of tilt & side length

- Back-up idea for combined changes:
          - as {{corners and sides}} change. 
          - as {{opposite}} {{corners and sides}} change {{unequally}}.

**Comment by jessegreenberg** [comment]
Notes:
- I am finding that the current regions may be too large (at least when using a keyboard, it may feel better when using the tangible).
- Often "straightening" a parallelogram results in larger changes to length than angle and so where we might expect to describe tilting/straightening we actually describe the changing area.

**Comment by terracoda** [comment]
Ok, we have slightly tweaked the rules or logic for our transition-type context responses:

### Timing of transition responses
- For every angle changes of 15 degrees, voice appropriate transition context response
- For every visual side length mark change, voice appropriate transition context response
- Basic context responses over-write transition context responses when they collide

### Tilt Parameters
**When maintaining a parallelogram the sides can tilt or straighten (squares and rectangles) in parallel as the angles change equally**
- Opposite sides {{noTiltChange}} {{in}} parallel + as + area changes 
- Opposite sides {{tilt}} {{in}} parallel}} + as + opposite corners change
- Opposite sides {{straighten}} {{in}} parallel + as + opposite corners change 

_When NOT maintaining a parallelogram we can track the true-ness of the parallel of the sides - not the direction of the lean of the other 2 sides, so:_

**When one pair of opposite sides move into a parallel state (i.e., finding a trapezoid) we describe the sides that are in parallel & a corner change since it is new that the sides have become parallel**
- Opposite sides {{sideName}} and {{sideName}} tilt {{in}} parallel + as corners change {{unequally}}.

**When one pair of opposite sides remain in parallel (i.e., maintaining a trapezoid) we keep describing the sides that are in parallel (ignoring the tilt of the sides that are not in parallel), & in add the area change as two sides remain in parallel and the shape changes size.** 
- Opposite sides {{sideName}} and {{sideName}} tilt {{in}} parallel + as shape gets {{bigger/smaller}}.

**When all sides lose their parallel state (like finding a kite, general quad or concave quad in one go) we describe that all sides tilt away from parallel & add the corner detail because we lost one pair of parallel sides:**
- {{All}} sides tilt {{away from}} parallel + as opposite corners change {{unequally}}.

**When transformations maintain shapes where no sides are parallel we continue describing that all sides tilt away from parallel & add the area change if that makes sense to the model**
- {{All}} sides tilt {{away from}} parallel + as shape gets {{bigger/smaller}}.

**When both sets of opposite sides move into parallel we describe one of the found a parallelogram responses, for example:** 
- {{Found a parallelogram}}.

### Corner change parameters
- + as {{opposite}} corners change {{equally/unequally}}
- For maintaining parallelograms the model can determine when angle changes are significant
- For non-parallelograms, I think the corner change is mostly used when a new state of parallel-ness occurs or is lost, for example finding or losing any kind of trapezoid

### Area change parameters
- + as shape gets {{smaller/bigger}}}.
- For maintaining parallelograms the model can determine when side-length is significant
- For non-parallelograms, I think the area change is mostly used when maintaining a shape where there is no change to the state of parallelism of any pair of opposite sides - no loss of parallelism and no finding of parallelism.

### Back-up idea for combined changes if we have issues with the design implementation or feedback from learners:
- +as {{corners and sides}} change. 
- + as {{opposite}} {{corners and sides}} change {{unequally}}.

**Comment by jessegreenberg** [comment]
OK I think this is in place @terracoda, ready for you to test and review. Hopefully it is working as expected, I was able to reproduce each example in the tables.

**Comment by terracoda** [comment]
I think you got everything in there @jessegreenberg! It sound really great to me.

I did find a case where I can move the sides of a trapezoid out of parallel and the response does not adjust to "All sides tilt out of parallel..."

**To reproduce:**
- start the a small trapezoid 
- grab and move a non-parallel side
- you should here that "opposite sides tilt" in parallel even when they are mo longer in parallel.

The description adjustment seems to work fine with larger shapes.

**Comment by jessegreenberg** [comment]
Thanks @terracoda, confirmed that case locally as well. Ill take a look.

**Comment by terracoda** [comment]
One other thing we will want to keep our ears and eyes on is how to best support the "other shape clicks".

For example, I am wondering if there is anyway to ensure that when a click makes any trapezoid, can we ensure that the latest response contains the "opposite sides in parallel" phrase?

A couple of times I noticed that I moved into a trapezoid, but the last response I heard was the phrase containing, "All sides tilt out of parallel...". 

I don't want to complicate things too much. I am just wonder if there is a way to get the latest response a key points like the one just mentioned.

**Comment by jessegreenberg** [comment]
> I did find a case where I can move the sides of a trapezoid out of parallel and the response does not adjust to "All sides tilt out of parallel..."

OK, it looks like this is a model related to https://github.com/phetsims/quadrilateral/issues/108. It is possible to set the tolerances such that sides that are not close to parallel are reported as such. Ill make a new issue for this.

<details>

![image](https://user-images.githubusercontent.com/6396244/169879939-8f2c1a76-271a-47c9-9b94-18b945883842.png)


</details>

**Comment by terracoda** [comment]
Regarding the other issue (https://github.com/phetsims/quadrilateral/issues/154#issuecomment-1134970775) about the shape sounds...I think that should do as you suggested and play a context response every time a new named shape is created? 

I think that would be our best starting point. We might then have to optimize the responses if we find that they are too long in those shape making cases.

**Comment by jessegreenberg** [comment]
@terracoda and I met on zoom to discuss in more detail and I think this change has been made. @terracoda let me know what you think.

**Comment by terracoda** [comment]
@jessegreenberg , I think the context responses are sounding good.

**Comment by terracoda** [comment]
I think we can close this issue. I have a question for design meeting about whether we need a special response to indicate 2 parallel sides. I'll put that in a new issue.
